repose-aggregator/components/filters/valkyrie-authorization/src/test/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilterTest.scala

```diff
@@ -177,14 +179,46 @@ class ValkyrieAuthorizationFilterTest extends FunSpec with BeforeAndAfter with M
+      (RequestProcessor("HEAD", Map("X-Tenant-Id" -> "hybrid:someTenant", "X-Contact-Id" -> "123456"), "http://foo.com:8080/foo"), ValkyrieResponse(200, createValkyrieResponse("123456", "view_product"))), //Without colon in tenant
+      (RequestProcessor("POST", Map("X-Tenant-Id" -> "hybrid:someTenant", "X-Contact-Id" -> "123456"), "http://foo.com:8080/foo"), ValkyrieResponse(200, createValkyrieResponse("123456", "edit_product"))), //Edit role
+      (RequestProcessor("PUT", Map("X-Tenant-Id" -> "hybrid:someTenant", "X-Contact-Id" -> "123456"), "http://foo.com:8080/foo"), ValkyrieResponse(200, createValkyrieResponse("123456", "admin_product"))), //Admin role
+      (RequestProcessor("GET", Map("X-Tenant-Id" -> "hybrid:someTenant", "X-Contact-Id" -> "123456"), "http://foo.com:8080/bar"), ValkyrieResponse(200, createValkyrieResponse("123456", "view_product"))), //View role
+      (RequestProcessor("HEAD", Map("X-Tenant-Id" -> "hybrid:someTenant", "X-Contact-Id" -> "123456"), "http://foo.com:8080/bar"), ValkyrieResponse(200, createValkyrieResponse("123456", "view_product"))), //Without colon in tenant
+      (RequestProcessor("POST", Map("X-Tenant-Id" -> "hybrid:someTenant", "X-Contact-Id" -> "123456"), "http://foo.com:8080/bar"), ValkyrieResponse(200, createValkyrieResponse("123456", "edit_product"))), //Edit role
+      (RequestProcessor("PUT", Map("X-Tenant-Id" -> "hybrid:someTenant", "X-Contact-Id" -> "123456"), "http://foo.com:8080/bar"), ValkyrieResponse(200, createValkyrieResponse("123456", "admin_product"))) //Admin role
+    ).foreach { case (request, valkyrie) =>
+      it(s"should allow requests for $request with Valkyrie response of $valkyrie without device id when on either accepted list") {
+        val akkaServiceClient: AkkaServiceClient = generateMockAkkaClient("someTenant", request.headers.getOrElse("X-Contact-Id", "ThisIsMissingAContact"), valkyrie.code, valkyrie.payload)
+
+        val filter: ValkyrieAuthorizationFilter = new ValkyrieAuthorizationFilter(mock[ConfigurationService], akkaServiceClient, mockDatastoreService)
+        filter.configurationUpdated(createGenericValkyrieConfiguration(null))
+
+        val mockServletRequest = new MockHttpServletRequest
+        mockServletRequest.setMethod(request.method)
+        mockServletRequest.setRequestURL(request.url)
+        request.headers.foreach { case (k, v) => mockServletRequest.setHeader(k, v) }
+
+        val mockFilterChain = mock[FilterChain]
+        val responseCaptor = ArgumentCaptor.forClass(classOf[MutableHttpServletResponse])
+        Mockito.when(mockFilterChain.doFilter(Matchers.any(classOf[ServletRequest]), responseCaptor.capture())).thenAnswer(new Answer[Unit] {
+          override def answer(invocation: InvocationOnMock): Unit = responseCaptor.getValue.getOutputStream.print(createOriginServiceResponse("98765", "123456"))
```

Comment:
yeah, this test grew a little to much without re looking at how things were being accomplished.